filament/src/components/RenderableManager.cpp

```diff
@@ -237,6 +265,109 @@ RenderableManager::Builder& RenderableManager::Builder::globalBlendOrderEnabled(
     return *this;
 }
 
+UTILS_NOINLINE
+void RenderableManager::BuilderDetails::processBoneIndicesAndWights(Engine& engine, Entity entity) {
+    size_t maxPairsCount = 0; //size of texture, number of bone pairs
+    size_t maxPairsCountPerVertex = 0; //maximum of number of bone per vertex
+
+    for (auto iBonePair = mBonePairs.begin(); iBonePair != mBonePairs.end(); ++iBonePair){
+        auto primitiveIndex = iBonePair->first;
+        auto entries = mEntries;
+        ASSERT_PRECONDITION(primitiveIndex < entries.size() && primitiveIndex >= 0,
+            "[primitive @ %u] primitiveindex is out of size (%u)", primitiveIndex, entries.size());
+        auto entry = mEntries[primitiveIndex];
+        auto bonePairsForPrimitive = iBonePair->second;
+        auto vertexCount = entry.vertices->getVertexCount();
+        ASSERT_PRECONDITION(bonePairsForPrimitive.size() == vertexCount,
+            "[primitive @ %u] bone indices and weights pairs count (%u) must be equal to vertex count (%u)",
+            primitiveIndex, bonePairsForPrimitive.size(), vertexCount);
+        auto const& declaredAttributes = downcast(entry.vertices)->getDeclaredAttributes();
+        ASSERT_PRECONDITION(!declaredAttributes[VertexAttribute::BONE_INDICES],
+            "[entity=%u, primitive @ %u] vertex attribute BONE_INDICES for skinning are already defined",
+            entity.getId(), primitiveIndex);
+        ASSERT_PRECONDITION(!declaredAttributes[VertexAttribute::BONE_WEIGHTS],
+            "[entity=%u, primitive @ %u] vertex attribute BONE_WEIGHTS for skinning are already defined",
+            entity.getId(), primitiveIndex);
+        for (size_t iVertex = 0; iVertex < vertexCount; iVertex++) {
+            auto bonesPerVertex = bonePairsForPrimitive[iVertex].size();
+            maxPairsCount +=  bonesPerVertex;
+            maxPairsCountPerVertex = max(bonesPerVertex, (uint) maxPairsCountPerVertex);
+        }
+    }
+
+    size_t pairsCount = 0; // counting of number of pairs stored in texture
+    if (maxPairsCount) { // at least one primitive has bone indices and weights
+        mBoneIndicesAndWeight = (float2*) calloc(maxPairsCount,
+            sizeof(float2)); // final texture data, indices and weights
```

Comment:
seems like `mBoneIndicesAndWeight` is leaked.

Reply:
fixed